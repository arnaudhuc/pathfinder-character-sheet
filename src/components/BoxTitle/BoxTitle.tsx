/** @jsx jsx */
import { jsx } from '@emotion/core';

interface IBoxTitle {
	title: string;
}

export const BoxTitle: React.FC<IBoxTitle> = (props) => {
	const { title } = props;
	return (
		<div
			css={{
				background: 'black',
				color: 'white',
				height: '30px',
				textAlign: 'center',
				display: 'flex',
				alignItems: 'center',
				justifyContent: 'center',
				textTransform: 'uppercase',
				fontSize: '20px',
			}}
		>
			{title}
		</div>
	);
};
