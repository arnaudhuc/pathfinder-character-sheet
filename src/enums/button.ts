export enum EColor {
	DEFAULT = 'default',
	PRIMARY = 'primary',
	SECONDARY = 'secondary',
	INHERIT = 'inherit',
}
