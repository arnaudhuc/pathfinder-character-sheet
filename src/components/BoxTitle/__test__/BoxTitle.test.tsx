/** @jsx jsx */
import { jsx } from '@emotion/core';
import renderer from 'react-test-renderer';

import { BoxTitle } from '../BoxTitle';
import { EColor } from '../../../enums/button';

describe('BoxTitle', () => {
	test('should render with default props', () => {
		const component = renderer.create(<BoxTitle title="Valeur" />).toJSON();

		expect(component).toMatchSnapshot();
	});

	test('should render with secondary color', () => {
		const component = renderer
			.create(<BoxTitle title="Valeur" color={EColor.SECONDARY} />)
			.toJSON();

		expect(component).toMatchSnapshot();
	});
});
