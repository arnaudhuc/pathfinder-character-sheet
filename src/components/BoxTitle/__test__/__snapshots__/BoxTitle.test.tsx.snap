// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxTitle should render with default props 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
    >
      Valeur
    </h2>
  </div>
</header>
`;

exports[`BoxTitle should render with secondary color 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorSecondary mui-fixed MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
    >
      Valeur
    </h2>
  </div>
</header>
`;
